src/components/Reports/ReportsTrafficSources/ReportsTrafficSources.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage, injectIntl, FormattedNumber } from 'react-intl';
+import Loading from '../../Loading/Loading';
+
+const ReportsTrafficSources = function({
+  domainName,
+  dateFrom,
+  intl,
+  dependencies: { datahubClient },
+}) {
+  const [trafficSources, setTrafficSources] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      const trafficSourcesData = await datahubClient.getTrafficSourcesByPeriod(
+        domainName,
+        dateFrom,
+      );
+      if (trafficSourcesData.success && trafficSourcesData.value) {
+        setTrafficSources(trafficSourcesData.value);
+        const total = trafficSources.reduce(function(previous, item) {
+          return previous + item.quantity;
+        }, 0);
+        setTotalQuantity(total);
+      }
+      setIsLoading(false);
+    };
+
+    fetchData();
+  }, [datahubClient, dateFrom, domainName, trafficSources]);
+
+  return (
+    <div className="wrapper-ranking">
+      <div className="reports-box">
+        <h4 className="title-ranking">
+          <FormattedMessage id="trafficSources.title" />
+        </h4>
+        <div>
+          {isLoading ? (
+            <Loading />
+          ) : trafficSources ? (
+            trafficSources.map((trafficSource, index) => (
+              <div key={index}>
+                <span>{trafficSource.sourceName}</span>
+                <span>
+                  {trafficSource.quantity} (
+                  <FormattedNumber value={(trafficSource.quantity * 100) / totalQuantity} />
```

Comment:
What do you think about specify how many decimals display?

Something like:

```suggestion
                  <FormattedNumber value={(trafficSource.quantity * 100) / totalQuantity} format={{ style: 'percent', minimumFractionDigits: 2, maximumFractionDigits: 2 }} />
```

See more details in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat

Reply:
doesn't work this

Reply:
fixed